torch/utils/_python_dispatch.py

```diff
@@ -66,6 +66,10 @@ def _get_current_dispatch_mode_stack():
     stack_len = _len_torch_dispatch_stack()
     return [_get_dispatch_stack_at(i) for i in range(stack_len)]
 
+def _is_mode_in_stack(modeCls):
+    stack_len = _len_torch_dispatch_stack()
+    return modeCls in [type(_get_dispatch_stack_at(i)) for i in range(stack_len)]
```

Comment:
Would like a little test for this.

This test is quadratic, maybe we can raise a warning if stack length is > 10 or something. One way to make it not require a mode stack scan is to maintain TLS on all mode objects saying if it's on the stack or not; then you just check your mode to see if the TLS is true or not.

A better version of this test would be to test it for ALL modes, not just fake tensor mode.

Reply:
This test isn't quite correct. It should take a mode object, and test if the mode object **is** on the stack. Some modes may be able to handle having multiple instances pushed onto the stack simultaneously (e.g., imagine some functorch style transform, where a transform can be applied multiple times.)

Reply:
I see, yes, if we want to do this for all modes, then yes, this is a better approach. Is this a valid constraint for all modes?

Reply:
Well, some modes may only want to be pushed once (or maybe are even singleton). But that has to be a per-mode policy decision. Maybe you could default modes to be pushed once only, with an opt-out to allow them to have multiple active instances. Maybe @samdow has an opinion.